Desktop/widgets/qmllistview.cpp

```diff
@@ -292,6 +386,65 @@ void QMLListView::setTermsAreInteractions()
 	model()->setTermsAreInteractions(true);
 }
 
+QList<std::pair<QMLListView::SourceType*, Terms> > QMLListView::getTermsPerSource()
+{
+	QList<std::pair<SourceType*, Terms> > result; // Do not use a map in order to keep the order of the sources.
+
+	for (SourceType* sourceItem : _sourceModels)
+	{
+		ListModel* sourceModel = sourceItem->model;
+		if (sourceModel)
+		{
+			Terms sourceTerms = sourceModel->terms(sourceItem->modelUse);
+
+			for (const SourceType& discardModel : sourceItem->getDiscardModels())
+				sourceTerms.discardWhatDoesContainTheseComponents(discardModel.model->terms(discardModel.modelUse));
+
+			if (!sourceItem->conditionExpression.isEmpty())
+			{
+				Terms filteredTerms;
+				QJSEngine jsEngine;
```

Comment:
Would it be nice to have this static?
Or is the initialization of QJSEngine light enough that we don't need to care?

Reply:
Another option would be to just use the QQmlEngine we already have running?
https://doc.qt.io/qt-5/qqmlengine.html shows me it inherits QJSEngine so it should be able to do everything we want and because QQMLListView is already a QML element it should already have access to it I think.